dds/DCPS/DataWriterImpl.cpp

```diff
@@ -1345,7 +1345,7 @@ DataWriterImpl::enable()
                                            qos_.reliability.max_blocking_time,
                                            n_chunks_,
                                            domain_id_,
-                                           get_topic_name(),
+                                           topic_name_.in(),
```

Comment:
The code is clear and simpler without .in() so that's what we prefer in OpenDDS.  I've never seen a problem with a compiler not allowing this.  Primarily we want users and developers to read and understand the code without getting bogged down in details of the IDL to C++ mapping.  We have some notes on this on an internal wiki page, I'll try moving them to the GH wiki.